cmd/contour/contour.go

```diff
@@ -120,7 +129,67 @@ func main() {
 			log.WithError(err).Fatal("invalid configuration")
 		}
 
-		if err := doServe(log, serveCtx); err != nil {
+		// Build out serve deps.
+		serve, err := NewServer(log, serveCtx)
+		if err != nil {
+			log.WithError(err).Fatal("unable to initialize Server dependencies required to start Contour")
+		}
+
+		var contourConfiguration contour_api_v1alpha1.ContourConfigurationSpec
+
+		// Get the ContourConfiguration CRD if specified
+		if len(serveCtx.contourConfigurationName) > 0 {
+			// Determine the name/namespace of the configuration resource utilizing the environment
+			// variable "CONTOUR_NAMESPACE" which should exist on the Contour deployment.
+			//
+			// If the env variable is not present, it will default to "projectcontour".
+			contourNamespace, found := os.LookupEnv("CONTOUR_NAMESPACE")
+			if !found {
+				contourNamespace = "projectcontour"
+			}
+
+			namespacedName := types.NamespacedName{Name: serveCtx.contourConfigurationName, Namespace: contourNamespace}
+			client := serve.clients.DynamicClient().Resource(contour_api_v1alpha1.ContourConfigurationGVR).Namespace(namespacedName.Namespace)
+
+			// ensure the specified ContourConfiguration exists
+			res, err := client.Get(context.Background(), namespacedName.Name, metav1.GetOptions{})
+			if err != nil {
+				log.WithError(err).Fatalf("error getting contour configuration %s", namespacedName)
+			}
+
+			var contourConfig contour_api_v1alpha1.ContourConfiguration
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(res.Object, &contourConfig); err != nil {
+				log.WithError(err).Fatalf("error converting contour configuration %s", namespacedName)
+			}
+
+			// Set valid status on the ContourConfiguration object
+			contourConfig.Status = contour_api_v1alpha1.ContourConfigurationStatus{
+				Conditions: []contour_api_v1.DetailedCondition{{
+					Condition: contour_api_v1.Condition{
+						Type:               "Valid",
+						Status:             "True",
+						ObservedGeneration: 0,
```

Comment:
This should be set to whatever `metadata.generation` is on the `contourConfig` object. Not a huge deal, but that's what the `observedGeneration` says: I based this Condition on this `metadata.generation` of the object.